### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by subhaklp**
1. Implemented a separate PermissionsNavTree.jsx file. Modified Sidebar to it either displays NavTree or PermissionsNavTree based on the location.
2. A memoized selector is in place under selectors/index.js to get us visible boundaries and boundary details. Needs some refactoring in terms of names etc..
3. Sidebar implements a mapStateToProps and calls this memoized selector to get a filtered list of boundaries and boundary details and passes it into PermissionsNavTree. 

Concerns:

- @aksanoble need you to just have a second pair of eyes at the logic and point out holes. 
- Please review the selectors and scan for logic errors
- Permissions Nav Tree has some weird arrows showing up in the UI. Good for now because it differentiates it from the regular nav tree but we don't really want it.. 👎  I don't know what's making it show up.

Need an overall code review before I merge it in. Lots of work still needed before the whole deal works..

### CODE REVIEW

File: app/components/SecondaryNavBar.js [hunk 1]
```diff
@@ -29,14 +29,19 @@ export default class SecondaryNavBar extends React.Component {
     this.props.redirectTo('/users');
   }
 
+  handleManagePermissionsBtnClick()
+  {
+    this.props.redirectTo('/permissions');
```
Comment (aksanoble) [inline app/components/SecondaryNavBar.js]: As we're using react-router-redux we want to do route transitions via call to the store. For example. `dispatch(push('/login'))`

File: app/components/SecondaryNavBar.js [hunk 2]
```diff
@@ -29,14 +29,19 @@ export default class SecondaryNavBar extends React.Component {
...
+
   render() {  
     var Displayelement;
     if(sessionStorage.getItem('isAdmin')) {
       Displayelement = (props) => {
          return (<div className="pull-right">
           <button type="button" className="btn btn-info navbar-btn all-padded-btn" data-toggle="tooltip" onClick={this.handleProgramClick}><span className="fa fa-pencil-square-o"></span> Manage Programs</button>
           <button type="button" className="btn btn-info navbar-btn all-padded-btn" onClick={ this.props.toggleDistrictModal } data-toggle="tooltip" data-placement="bottom" title="Create District"><span className="fa fa-globe"></span></button>
-          <button type="button" className="btn btn-info navbar-btn all-padded-btn" data-toggle="tooltip" data-placement="bottom" title="Manage Permissions"><span className="fa fa-key"></span></button>
+          <button type="button" className="btn btn-info navbar-btn all-padded-btn" data-toggle="tooltip" data-placement="bottom" title="Manage Permissions" onClick={this.handleManagePermissionsBtnClick.bind(this)}><span className="fa fa-key"></span></button>
```
Comment (aksanoble) [inline app/components/SecondaryNavBar.js]: The name sounds a bit clunky. How about just `managePermissions`.

File: app/components/SideBar.js [hunk 3]
```diff
@@ -1,51 +1,88 @@
...
     const schoolType = primarySelected ? 1 : 2
-    boundariesByParentId[1] = _.filter(boundariesByParentId[1], (key) =>  {
+    boundariesByParentId[1] = _.filter(boundariesByParentId[1], (key) => {
       const boundaryType = boundaryDetails[key].boundary_type
       return boundaryType ? boundaryType == schoolType : true
     })
     var sidebarClass = classNames({
       'toggled': this.state.isExpanded
     })
+    if(location.pathname.includes("permissions"))
```
Comment (aksanoble) [inline app/components/SideBar.js]: Ideally we'd want our UI to be a function of state in the store or localState. So one way to do that could be add a `navTreeType` to our redux store. Set NavTree type `onEnter` or `componentDidMount` on the `permissions` route. And again reset on the `boundary` routes.

File: app/components/SecondaryNavBar.js (hunk 1 above)
Comment (subhaklp) [reply app/components/SecondaryNavBar.js]: @aksanoble the redirectTo method eventually does that..this is just a wrapper around push in the App.js file..

File: app/components/SecondaryNavBar.js (hunk 2 above)
Comment (subhaklp) [reply app/components/SecondaryNavBar.js]: Sure, will change..

File: app/components/SideBar.js (hunk 3 above)
Comment (subhaklp) [reply app/components/SideBar.js]: hmm..ok, i'll work on this..

File: app/components/SecondaryNavBar.js (hunk 2 above)
Comment (subhaklp) [reply app/components/SecondaryNavBar.js]: Done

**Pull request merged**